### DISCUSSION TITLE
Why aren't files completed for ... ?

### DISCUSSION THREAD

**Discussion by rsteube** [carapace-sh/carapace-bin#1655] [Q&A] [answered]
When trying to complete command ... files should be shown, but they aren't.

**Comment by rsteube** [comment, marked as answer]
This can have several reasons:

1. **The command is misconfigured/incomplete.**
Contrary to many shells file completion needs to be explicitly defined in carapace.
Having a file completion fallback helps when the shell completion is incomplete but most of  the time gives invalid suggestions.

2. **File completion is dependent on a flag or positional argument.**
Some commands have different behaviour depending on flags or positional arguments.
When file completion would be invalid it is omitted.

**Reply by BatmanAoD** [reply]
Is there any way to configure Carapace to fall-back to file-based completion for all unrecognized commands?

**Reply by rsteube** [reply]
Not as generic fall-back, no. But there is [overlay](https://rsteube.github.io/carapace-bin/overlay.html) as a workaround for missing commands.

**Reply by robrecord** [reply]
how would you suggest this is used? create a new overlay for every missing command?

**Reply by rsteube** [reply]
You can define multiple subcommands in an overlay.
